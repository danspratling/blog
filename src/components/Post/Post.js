/** @jsx jsx */
import { jsx } from 'theme-ui'
import { Link } from 'gatsby'

const Post = ({ node }) => {
  const randNum = (min, max) => Math.floor(Math.random() * (max - min)) + min

  const generateGradient = () => {
    const hue = randNum(0, 255)
    const sat = randNum(75, 100)
    const lit = 75

    return `linear-gradient(to right,
      hsl(
        ${hue - 15},
        ${sat}%,
        ${lit}%
      ),
      hsl(
        ${hue + 15},
        ${sat}%,
        ${lit}%
      ))`
  }

  return (
    <article
      key={node.fields.slug}
      sx={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        borderRadius: 5,
        backgroundImage: generateGradient(),
        border: '5px solid #ffffff',
        mb: 3,
        transition: '0.3s ease',
        ':hover, :focus, :active': {
          mt: -10,
        },
      }}
    >
      <Link
        to={node.fields.slug}
        sx={{
          variant: 'links.primary',
          padding: 4,
          boxShadow: `none`,
        }}
      >
        <header>
          <h3 sx={{ my: 2 }}>{node.frontmatter.title || node.fields.slug}</h3>
          <small>{node.frontmatter.date}</small>
        </header>
        <section>
          <p
            sx={{
              mb: 0,
            }}
            dangerouslySetInnerHTML={{
              __html: node.frontmatter.description || node.excerpt,
            }}
          />
        </section>
      </Link>
    </article>
  )
}

export default Post
