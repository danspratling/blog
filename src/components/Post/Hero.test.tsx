/** @jsx jsx */
import { jsx } from 'theme-ui'
import { ThemeProvider, useThemeUI } from 'theme-ui'
import renderer from 'react-test-renderer'
import Post from './Post'

const data = {
  excerpt: 'the excerpt',
  fields: {
    slug: 'http://example.com',
  },
  frontmatter: {
    title: 'the title',
    description: 'the description',
    date: '22/06/1994',
  },
}

describe('Post', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        // <ThemeProvider theme={theme}>
        <Post node={data} />
        // </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
