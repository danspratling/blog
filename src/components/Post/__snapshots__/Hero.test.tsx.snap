// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<article
  className="css-105sco2-Post"
>
  <a
    className="css-1ram7fh-Post"
    href="http://example.com"
  >
    <h3
      className="css-tsb5c-Post"
    >
      the title
    </h3>
    <small>
      22/06/1994
    </small>
    <p
      className="css-yl6mxk-Post"
      dangerouslySetInnerHTML={
        Object {
          "__html": "the description",
        }
      }
    />
  </a>
</article>
`;
